openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java

```diff
@@ -201,16 +201,16 @@ public static Schema<?> getOpenApiSchema(SyntaxKind type) {
     /**
      * Generate operationId by removing special characters.
      *
-     * @param identifier input function name, record name or operation Id
+     * @param operationID input function name, record name or operation Id
      * @return string with new generated name
      */
-    public static String getValidName(String identifier) {
+    public static String getOperationId(String operationID) {
         //For the flatten enable we need to remove first Part of valid name check
-        // this - > !identifier.matches("\\b[a-zA-Z][a-zA-Z0-9]*\\b") &&
-        if (!identifier.matches("\\b[0-9]*\\b")) {
-            String[] split = identifier.split(SPECIAL_CHAR_REGEX);
+        // this - > !operationID.matches("\\b[a-zA-Z][a-zA-Z0-9]*\\b") &&
+        if (!operationID.matches("\\b[0-9]*\\b")) {
```

Comment:
Shall we do an early return in here? Otherwise the string processing operation in L224 will be applied even if the operationId is fully numerical.

Reply:
Yeah , I will change